Xamarin.Forms.Platform.Android/Platform.cs

```diff
@@ -1134,17 +1134,23 @@ protected override int GetChildDrawingOrder(int childCount, int i)
 			{
 				//On Material design the button states use Elevation property, we need to make sure
 				//we update the elevation of other controls to be over the previous one
-				if (!_minimumElevation.ContainsKey(i))
-					_minimumElevation[i] = GetChildAt(i).Elevation;
-				for (int j = 0; j < _minimumElevation.Count() - 1; j++)
+				if (Forms.IsLollipopOrNewer)
 				{
-					while (_minimumElevation[j] > _minimumElevation[j + 1])
+					if (!_minimumElevation.ContainsKey(i))
```

Comment:
For this to work, shouldn't we remove this `if` statement? Then when we get a new index, we whack it's elevation into `_minimumElevation`. Then we re-establish strict ordering of `_minimumElement` in the loops below? Then if we don't have too many indices (so we're not worried about the perf of the loops below) then this will work.

Or if `_minimumElevation` already contains the key will the key's value will be the same as the new value? Put another way, will the new elevation always be the same as the old elevation for a given `i`? If so we should assert that fact and forgo the loops below (as we do not modify `_minimumElevation`).